dagshub/models/model_locator.py

```diff
@@ -0,0 +1,320 @@
+import logging
+import os
+from dataclasses import dataclass
+from enum import auto, IntEnum
+from functools import cached_property
+from pathlib import Path, PurePosixPath
+from typing import Optional, Literal, Union, Tuple, List
+
+import dacite
+import yaml
+from pathvalidate import sanitize_filepath
+
+from dagshub.common.analytics import send_analytics_event
+from dagshub.common.api import RepoAPI
+from dagshub.common.api.repo import PathNotFoundError
+from dagshub.common.api.responses import StorageAPIEntry
+from dagshub.common.determine_repo import determine_repo
+from dagshub.common.helpers import log_message
+from dagshub.models.model_loaders import ModelLoader, RepoModelLoader, BucketModelLoader, DagsHubStorageModelLoader
+
+# TODO:
+"""
+Ways to get model:
+- automatic - with the cascade
+- from path
+- from bucket
+- from mlflow model
+- from mlflow artifact
+"""
+
+# TODO: don't forget to separate dvc downloads by the hash of the dir
```

Comment:
Is this intention for future or for this PR?

Reply:
For the future.
The idea is to actually redownload the model if there's a newer version in the repo (unless a revision was specified).
If we download the model into the hash of the dir, we can do that "easily" by separating the model versions every time.

I didn't really consider how this will work with e.g. buckets, so this is for now left as a "do this in the future"